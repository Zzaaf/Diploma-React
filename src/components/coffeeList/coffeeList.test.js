import React from 'react';
import {shallow} from 'enzyme';
import CoffeeList from '../coffeeList';

describe('Testing <CoffeeList/>', () => {
    const items = shallow(<CoffeeList/>);
    describe('Testing snap & state', () => {
        it('CoffeeList has rendered correctly', () => {
            expect(items).toMatchSnapshot();
        });
        it('CoffeeList state "posts" is array', () => {
            expect(items.state().posts).toBeArray();
        });
        it('CoffeeList state "loading" is true', () => {
            expect(items.state().loading).toBeTruthy();
        });
        it('CoffeeList state "typeError" is string', () => {
            expect(items.state().typeError).toBeString();
        });
        it('CoffeeList state "fatalError" is false', () => {
            expect(items.state().fatalError).toBeFalsy();
        });
    });
});