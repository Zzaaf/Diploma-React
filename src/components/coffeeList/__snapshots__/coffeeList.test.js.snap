// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <CoffeeList/> Testing snap & state CoffeeList has rendered correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CoffeeList />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <Spinner />,
      "tag": "div",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Spinner />,
        "tag": "div",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
